### PR TITLE
HHH-11716 - @Transient annotation not respected when class defines 'get' and 'is' accessor variants

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-11716

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/internal/util/ReflectHelper.java [hunk 1]
```diff
@@ -451,6 +453,10 @@ private static Method getGetterOrNull(Class containerClass, String propertyName)
 				continue;
 			}
 
+			if (method.getAnnotation( Transient.class ) != null){
```
Comment (Naros) [inline hibernate-core/src/main/java/org/hibernate/internal/util/ReflectHelper.java]: Checkstyle fix required here.

**Comment by dreab8** [comment]
@Naros fixed
Thanks

**Comment by vladmihalcea** [comment]
Applied upstream, thanks

**Comment by peterjurkovic** [comment]
Would it be possible also to merge it into 5.0.X branch?

**Pull request closed without merging**